test/Microsoft.TemplateEngine.Utils.UnitTests/SemanticVersionTests.cs

```diff
@@ -0,0 +1,207 @@
+                Assert.True(SemanticVersion.TryParse(source, out ver));
+            }
+            else
+            {
+                Assert.False(SemanticVersion.TryParse(source, out ver));
+                return;
+            }
+
+            Assert.Equal(major, ver.Major);
+            Assert.Equal(minor, ver.Minor);
+            Assert.Equal(patch, ver.Patch);
+            Assert.Equal(prerelease, ver.PrereleaseInfo);
+            Assert.Equal(metadata, ver.BuildMetadata);
+        }
+
+        [Theory(DisplayName = nameof(SemanticVersionCompareTo))]
+        [InlineData("1.0.0", "1.0.0-beta1", true, false)]
+        [InlineData("1.0.0", "1.0.0-beta1", true, true)]
+        [InlineData("1.0.0", "1.0.0-beta1", null, false)]
+
+        [InlineData("1.0.0-alpha.100", "1.0.0-alpha.99", true, true)]
+        [InlineData("1.0.0-alpha.100", "1.0.0-alpha.99", true, false)]
+        [InlineData("1.0.0-alpha.100", "1.0.0-alpha.99", null, false)]
+
+        [InlineData("1.0.0-alpha.99.test", "1.0.0-alpha.99", true, true)]
+        [InlineData("1.0.0-alpha.99.test", "1.0.0-alpha.99", true, false)]
+        [InlineData("1.0.0-alpha.99.test", "1.0.0-alpha.99", null, false)]
+
+        [InlineData("1.0.0-alpha.99", "1.0.0-alpha.test", false, false)]
+        [InlineData("1.0.0-alpha.99", "1.0.0-alpha.test", false, true)]
+        [InlineData("1.0.0-alpha.99", "1.0.0-alpha.test", null, false)]
+
+        [InlineData("1.0.0-alpha1", "1.0.0-beta1", false, false)]
+        [InlineData("1.0.0-alpha1", "1.0.0-beta1", false, true)]
+        [InlineData("1.0.0-alpha1", "1.0.0-beta1", null, false)]
+
+        [InlineData("1.0.0-beta1", "1.0.0-alpha1", true, false)]
+        [InlineData("1.0.0-beta1", "1.0.0-alpha1", true, true)]
+        [InlineData("1.0.0-beta1", "1.0.0-alpha1", null, false)]
+
+        [InlineData("1.0.0-alpha1", "1.0.0-alpha1", true, true)]
+        [InlineData("1.0.0-alpha1", "1.0.0-alpha1", false, true)]
+        [InlineData("1.0.0-alpha1", "1.0.0-alpha1", null, true)]
+
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build42", true, true)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build42", false, true)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build42", null, true)]
+
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-beta1+build42", false, false)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-beta1+build42", false, true)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-beta1+build42", null, false)]
+
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build43", false, false)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build43", false, true)]
+        [InlineData("1.0.0-alpha1+build42", "1.0.0-alpha1+build43", null, false)] // This is expected to differ from the Operators test - as build metada has to be accounted for when performing a comparison sort
+        public void SemanticVersionCompareTo(string ver1, string ver2, bool? greater, bool equal)
```

Comment:
Correct the failure messages to indicate the the comparisons that are actually being done, the current messages imply usage of the operators rather than checking the numeric result of `CompareTo(SemanticVersion)`

Reply:
This is done